addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,260 @@
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+    private static final String MESSAGE_PREFIX = "ascanalpha.xsltinjection.";
+
+    private static final String[] errorCausingPayloads = {"<"};
+
+    private static final String[] evidenceError = {
+        "compilation error", "XSLT compile error", "SAXParseException"
+    };
+
+    private static final String[] vendorReturningPayloads = {
+        "<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+        "system-property(\'xsl:vendor\')/>",
+        "\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+        "<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+    };
+
+    private static final String[] evidenceVendors = {
+        "libxslt", "Microsoft", "Saxonica", "Apache", "Xalan", "SAXON", "Transformiix"
+    };
+
+    private static final String[] evidencePortScanning = {
+        "failed to open stream",
+        "Invalid Http response",
+        "Connection Refused",
+        "No connection could be made because the target machine actively refused it",
+        "Can not load requested doc"
+    };
+
+    private static final String[] cmdExecPayloads = {
+        "<xsl:variable name=\"rtobject\" select=\"runtime:getRuntime()\"/>\n"
+                + "<xsl:variable name=\"process\" select=\"runtime:exec($rtobject,'erroneous_command')\"/>\n"
+                + "<xsl:variable name=\"waiting\" select=\"process:waitFor($process)\"/>\n"
+                + "<xsl:value-of select=\"$process\"/>",
+        "<xsl:value-of select=\"php:function('exec','erroneous_command 2>&amp;1')\"/>"
+    };
+
+    private static final String[] evidenceCmdExec = {
+        "Cannot run program", "erroneous_command: not found",
+    };
+
+    private enum XSLTInjectionType {
+        ERROR(errorCausingPayloads, evidenceError, "error"),
+        VENDOR(vendorReturningPayloads, evidenceVendors, "vendor"),
+        PORTSCAN(new String[] {}, evidencePortScanning, "portscan"),
+        COMMAND_EXEC(cmdExecPayloads, evidenceCmdExec, "command");
+
+        private String[] payloads;
+        private String[] responses;
+        private String resourceIdentifier;
+
+        XSLTInjectionType(String[] payloads, String[] responses, String resourceIdentifier) {
+            this.payloads = payloads;
+            this.responses = responses;
+            this.resourceIdentifier = resourceIdentifier;
+        }
+    }
+
+    private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+    // used to check against the attack strength
+    private int requestsSent = 0;
+    // map setting the limits associated with each attack strength
+    private static Map<AttackStrength, Integer> strengthToRequestCountMap = new HashMap<>();
+
+    static {
+        strengthToRequestCountMap.put(AttackStrength.LOW, 6);
+        strengthToRequestCountMap.put(AttackStrength.MEDIUM, 12);
+        strengthToRequestCountMap.put(AttackStrength.HIGH, 24);
```

Comment:
I did some testing this morning, the max currently seems to be 8 (logical given the sets of payloads). Let's restore INSANE with the same value as HIGH (for the time being).